python/tests/unit/test_session.py

```diff
@@ -0,0 +1,137 @@
+import pytest
+
+from saucebindings.options import SauceOptions
+from saucebindings.session import SauceSession
+
+
+class TestInit(object):
+
+    def test_defaults_to_us_west_data_center(self):
+        session = SauceSession()
+
+        assert "us-west-1" in session.remote_url
+
+    def test_overrides_default_value_for_data_center(self):
+        session = SauceSession(data_center='eu-central')
+
+        assert "eu-central-1" in session.remote_url
+
+    def test_raises_exception_if_data_center_is_invalid(self):
+        with pytest.raises(ValueError):
+            SauceSession(data_center='invalid')
+
+    def test_accepts_provided_options_instance(self):
+        options = SauceOptions()
+
+        session = SauceSession(options)
+
+        assert session.options.browser_name == 'chrome'
+        assert session.options.browser_version == 'latest'
+        assert session.options.platform_name == 'Windows 10'
+
+    def test_generates_default_options_instance_if_not_provided(self):
+        session = SauceSession()
+
+        assert session.options.browser_name == 'chrome'
+        assert session.options.browser_version == 'latest'
+        assert session.options.platform_name == 'Windows 10'
+
+
+class TestDataCenter(object):
+
+    def test_overrides_default_value_for_data_center(self):
+        session = SauceSession()
+
+        session.data_center = 'eu-central'
+
+        assert "eu-central-1" in session.remote_url
+
+    def test_raises_exception_if_data_center_is_invalid(self):
+        session = SauceSession()
+
+        with pytest.raises(ValueError):
+            session.data_center = 'invalid'
+
+
+class TestStart(object):
+    @pytest.fixture(autouse=True)
```

Comment:
Don't need `autouse=True` since you're explicitly using the fixture in the test methods